src/error.rs

```diff
@@ -160,33 +163,11 @@ impl From<header::InvalidHeaderValueBytes> for Error {
     }
 }
 
-// A crate-private type until we can use !.
-//
-// Being crate-private, we should be able to swap the type out in a
-// backwards compatible way.
-pub enum Never {}
-
-impl From<Never> for Error {
-    fn from(never: Never) -> Error {
-        match never {}
-    }
-}
-
-impl fmt::Debug for Never {
-    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match *self {}
-    }
-}
-
-impl fmt::Display for Never {
-    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match *self {}
-    }
-}
-
-impl error::Error for Never {
-    fn description(&self) -> &str {
-        match *self {}
+impl From<std::convert::Infallible> for Error {
```

Comment:
Sean correctly pointed out that this might be a forwards-compatibility hazard once `!` lands in Rust.

Reply:
I'm curious how this may create forward-compatibility problems. [The docs](https://doc.rust-lang.org/std/convert/enum.Infallible.html#future-compatibility) only warn about implementing a trait for both Infallible *and* `!` but this does not appear to be the case here.

Reply:
Hmm, then I think this might be fine?